rcl/src/rcl/subscription.c

```diff
@@ -120,6 +170,22 @@ rcl_subscription_init(
     options->qos.avoid_ros_namespace_conventions;
   // options
   subscription->impl->options = *options;
+  subscription->impl->type_support = type_support;
+
+  if (options->rmw_subscription_options.content_filter_options) {
+    // Content filter topic not supported (or not enabled as some failed cases) on DDS.
+    // TODO(iuhilnehc-ynos): enable common content filter with an environment variable
+    // (e.g. FORCE_COMMON_CONTENT_FILTER) regardless of whether cft is enabled on DDS.
+    if (!subscription->impl->rmw_handle->is_cft_enabled) {
```

Comment:
what if rmw_create_subscription failed to enable CFT in rmw implementation? in that case, it will not fallback to this, right because rmw_create_subscription just fails? that means rmw CFT and rcl CFT have to be exclusive? i would expect, if rmw CFT fails to be enabled, then it will try to fall back to rcl CFT.

Reply:
> what if rmw_create_subscription failed to enable CFT in rmw implementation? in that case, it will not fallback to this, right because rmw_create_subscription just fails? 

No, we expect to fallback to rcl CFT if rmw CFT is failed, this is what this code snippet for.
If rmw CFT is failed, `rmw_create_subscription` should also return OK but with `subscription->impl->rmw_handle->is_cft_enabled` is false.

Reply:
yeah, that is my expectation, but implementation is not like that?

https://github.com/ros2/rmw_fastrtps/blob/1bf9e7a2d18caf6d50abf4aff1212b6d7dfc49f3/rmw_fastrtps_cpp/src/subscription.cpp#L247

and eventually `rmw_create_subscription` fails to return `NULL`.

Reply:
Thank you, I'll update it.